src/mrvi/_model.py

```diff
@@ -743,13 +771,122 @@ def get_aggregated_posterior(
             dist.Normal(qu_loc, qu_scale),
         )
 
+    def differential_abundance(
+        self,
+        adata: AnnData | None = None,
+        sample_cov_keys: list[str] | None = None,
+        sample_subset: list[str] | None = None,
+        batch_size: int = 128,
+    ) -> xr.Dataset:
+        """
+        Compute the differential abundance between samples.
+
+        Computes the logarithm of the ratio of the probabilities of each sample
+        conditioned on the estimated aggregate posterior distribution of each cell.
+
+        Parameters
+        ----------
+        adata
+            The data object to compute the differential abundance for.
+            If not given, the data object stored in the model is used.
+        sample_cov_keys
+            Keys for covariates (batch, etc.) that should also be taken into account
+            when computing the differential abundance.
+        sample_subset
+            Only compute differential abundance for these sample labels.
+        batch_size
+            Minibatch size for computing the differential abundance.
+
+        Returns
+        -------
+        :class:`xarray.Dataset`
+            Returns an xarray.Dataset with data variables:
+            - `log_probs`: Array of shape (n_cells, n_samples) containing the log probabilities for each cell across samples.
+            - `{cov_key}_log_probs`: For each key in `sample_cov_keys`, an array of shape (n_cells, n_cov_values) containing the log probabilities for each cell across covariate values.
+        """
```

Comment:
an alternative or a complement to this output could also be to return the log enrichment scores, of the form
log_prob[:, current_cov_value] - log_prob[:, np.arange(log_prob.shape[1]) != current_cov_value].mean(-1)
I feel like many people could also be interested in this output.